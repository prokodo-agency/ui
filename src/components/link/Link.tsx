"use client"
import { type FC, memo } from "react"

import { create } from "@/helpers/bem"
import { isString } from "@/helpers/validations"

import { BaseLink } from "../base-link"

import styles from "./Link.module.scss"

import type { LinkProps } from "./Link.model"

const bem = create(styles, "Link")

export const Link: FC<LinkProps> = memo(
  ({
    variant = "inherit",
    href,
    children,
    className,
    locale,
    style,
    target,
    itemProp,
    hasBackground,
    onMouseEnter,
    onClick,
    onKeyDown,
    ariaLabel,
  }) => {
    const linkModifier = {
      "has-no-background": hasBackground === false,
      [`has-no-background--${variant}`]: hasBackground === false,
    }
    const defaultProps = {
      className: bem(undefined, linkModifier, className),
      style,
      onMouseEnter,
    }
    if (onClick && !isString(href)) {
      return (
        <span
          aria-label={ariaLabel}
          role="button"
          tabIndex={0}
          onClick={onClick}
          onKeyDown={onKeyDown}
          {...defaultProps}
        >
          {children}
        </span>
      )
    }
    return (
      <BaseLink
        {...defaultProps}
        aria-label={ariaLabel}
        href={href}
        itemProp={itemProp}
        locale={locale}
        target={target ?? undefined}
        onClick={onClick}
      >
        {children}
      </BaseLink>
    )
  },
)

Link.displayName = "Link"
