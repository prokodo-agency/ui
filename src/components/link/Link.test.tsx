import { render, screen } from "@/tests"

import { Link } from "./Link"

describe("The common link component", () => {
  it("should render primary", () => {
    render(
      <Link href="/" variant="primary">
        Primary link
      </Link>,
    )
    expect(screen.getByText("Primary link")).toMatchSnapshot()
  })

  it("should render secondary", () => {
    render(
      <Link href="/" variant="secondary">
        Secondary link
      </Link>,
    )
    expect(screen.getByText("Secondary link")).toMatchSnapshot()
  })

  it("should not open inbound links in a new tab", () => {
    render(
      <Link href="/" variant="primary">
        Inbound link
      </Link>,
    )
    const inbound = screen.getByText("Inbound link")
    expect(inbound).not.toHaveAttribute("rel")
    expect(inbound).not.toHaveAttribute("target")
  })
})
