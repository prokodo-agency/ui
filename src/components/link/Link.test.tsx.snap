// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The common link component should render primary 1`] = `undefined`;

exports[`The common link component should render secondary 1`] = `undefined`;
